langchain/src/tools/google_calendar/base.ts

```diff
@@ -0,0 +1,91 @@
+import { google } from "googleapis";
+import { Tool } from "../base.js";
+import { OpenAI } from "../../llms/openai.js";
+import { getEnvironmentVariable } from "../../util/env.js";
+
+export interface GoogleCalendarAgentParams {
+  credentials: {
+    clientEmail?: string;
+    privateKey?: string;
+    calendarId?: string;
+  };
+  scopes?: string[];
+}
+
+export class GoogleCalendarBase extends Tool {
+  name = "Google Calendar";
+
+  description =
+    "A tool to lookup Google Calendar events and create events in Google Calendar";
+
+  protected clientEmail: string;
+
+  protected privateKey: string;
+
+  protected calendarId: string;
+
+  protected scopes: string[];
+
+  constructor(
+    fields: GoogleCalendarAgentParams = {
+      credentials: {
+        clientEmail: getEnvironmentVariable("GOOGLE_CALENDAR_CLIENT_EMAIL"),
+        privateKey: getEnvironmentVariable("GOOGLE_CALENDAR_PRIVATE_KEY"),
+        calendarId: getEnvironmentVariable("GOOGLE_CALENDAR_CALENDAR_ID"),
+      },
+      scopes: [
+        "https://www.googleapis.com/auth/calendar",
+        "https://www.googleapis.com/auth/calendar.events",
+      ],
+    }
+  ) {
+    super(...arguments);
+    if (!fields.credentials) {
+      throw new Error("Missing credentials to authenticate to Google Calendar");
+    }
+
+    if (!fields.credentials.clientEmail) {
+      throw new Error(
+        "Missing GOOGLE_CALENDAR_CLIENT_EMAIL to interact with Google Calendar"
+      );
+    }
+
+    if (!fields.credentials.privateKey) {
+      throw new Error(
+        "Missing GOOGLE_CALENDAR_PRIVATE_KEY to interact with Google Calendar"
+      );
+    }
+
+    if (!fields.credentials.calendarId) {
+      throw new Error(
+        "Missing GOOGLE_CALENDAR_CALENDAR_ID to interact with Google Calendar"
+      );
+    }
+
+    this.clientEmail = fields.credentials.clientEmail;
+    this.privateKey = fields.credentials.privateKey;
+    this.calendarId = fields.credentials.calendarId;
+    this.scopes = fields.scopes || [];
+  }
+
+  getModel() {
```

Comment:
This should be passed in

Reply:
This should accept a `BaseLLM` instance - what if someone needed to use a custom model?